package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,432 @@
+        pass
+
+
+class EpsilonConstant(Epsilon):
+    """Premade function for the determination of epsilon based on providing
+    an arbitrary constant"""
+
+    def __init__(self, DistanceMatrix, epsilon):
+        """
+        Parameters
+        ----------
+        epsilon : int
+        The value of epsilon to be used as a local scale parameter
+        """
+        if DistanceMatrix.calculated:
+            self.scaled_matrix = DistanceMatrix.dist_matrix
+        else:
+            raise AttributeError('Distance Matrix does not exist, was'
+                                 'DistanceMatrix.run() called?')
+
+        self._epsilon = epsilon
+        self.calculated = False
+
+    def determine_epsilon(self):
+        self.scaled_matrix /= self._epsilon
+        self.calculated = True
+        return
+
+class DiffusionMap(object):
+    """Non-linear dimension reduction method
+
+    Dimension reduction with diffusion mapping of selected structures in a
+    trajectory.
+
+    Attributes
+    ----------
+    eigenvalues: array
+        Eigenvalues of the diffusion map
+    eigenvectors: array
+        Eigenvectors of the diffusion map
+    diffusion_space : array
+        After calling `transform(num_eigenvectors)` the diffusion map embedding
+        into the lower dimensional diffusion space will exist here.
+
+    Methods
+    -------
+    run()
+        Constructs an anisotropic diffusion kernel and performs eigenvalue
+        decomposition on it.
+    transform(num_eigenvectors)
+        Perform an embedding of a frame into the eigenvectors representing
+        the collective coordinates.
+    """
+
+    def __init__(self, u, epsilon=1, manifold_density=None,
+                 timescale=1, force=False, **kwargs):
+        """
+        Parameters
+        -------------
+        u : MDAnalysis Universe or DistanceMatrix object
+            Can be a Universe, in which case one must supply kwargs for the
+            initialization of a DistanceMatrix. Otherwise, this can be a
+            DistanceMatrix already initialized. Either way, this will be made
+            into a diffusion kernel.
+        epsilon : Float
+            Specifies the method used for the choice of scale parameter in the
+            diffusion map. More information in [1], [2] and [3], Default: 1.
+        manifold_density: list, optional
+            The list has to have the same length as the trajectory.
+            With 'None' the weight of each frame of the trajectory will be the
+            same, Default : None
+        timescale: int, optional
+            The number of steps in the random walk, large t reflects global
+            structure whereas small t indicates local structure, Default: 1
+        force : boolean, optional
+            A boolean to override an exception thrown for a large
+            DistanceMatrix, Default: False
+        """
+        if isinstance(u, mda.Universe):
+            select = kwargs.get('select', "all")
```

Comment:
```
/home/jdetlefs/github/MDAnalysis/package/MDAnalysis/analysis/diffusionmap.py in __init__(self, u, epsilon, manifold_density, timescale, **kwargs)
    286             stop = kwargs.get('stop')
    287             step = kwargs.get('step')
--> 288             self._dist_matrix = DistanceMatrix(u,kwargs)
    289         elif isinstance(u, DistanceMatrix):
    290             self._dist_matrix = u

/home/jdetlefs/github/MDAnalysis/package/MDAnalysis/analysis/diffusionmap.py in __init__(self, u, select, metric, cutoff, weights, start, stop, step)
    196         self._u = u
    197         traj = self._u.trajectory
--> 198         self.atoms = self._u.select_atoms(select)
    199         self._metric = metric
    200         self._cutoff = cutoff

/home/jdetlefs/github/MDAnalysis/package/MDAnalysis/core/AtomGroup.pyc in select_atoms(self, sel, *othersel, **selgroups)
   4920         .. SeeAlso:: :meth:`AtomGroup.select_atoms` for more details.
   4921         """
-> 4922         return self.atoms.select_atoms(sel, *othersel, **selgroups)
   4923 
   4924     selectAtoms = deprecate(select_atoms,

/home/jdetlefs/github/MDAnalysis/package/MDAnalysis/core/AtomGroup.pyc in select_atoms(self, selstr, *othersel, **selgroups)
   3308            Added implicit OR syntax to field and range selections
   3309         """
-> 3310         atomgrp = Selection.Parser.parse(selstr, selgroups).apply(self)
   3311         # Generate a selection for each selection string
   3312         for sel in othersel:

/home/jdetlefs/github/MDAnalysis/package/MDAnalysis/core/Selection.pyc in parse(self, selectstr, selgroups)
    996         self.selectstr = selectstr
    997         self.selgroups = selgroups
--> 998         tokens = selectstr.replace('(', ' ( ').replace(')', ' ) ')
    999         self.tokens = collections.deque(tokens.split() + [None])
   1000         parsetree = self.parse_expression(0)
```
it looks like self._dist_matrix = DistanceMatrix(u, kwargs) thinks that the select kwargument is a dict and not a string.